slick.core.js

```diff
@@ -698,6 +698,62 @@
       });
     };
   }
+  
+  /***
+   * Polyfill for Map to support old browsers but
+   * benefit of the Map speed in modern browsers.
+   * @class Map
+   * @constructor
+   */
+  var Map = 'Map' in window ? window.Map : function Map() {
+    var data = {};
+    
+    /***
+     * Gets the item with the given key from the map or undefined if
+     * the map does not contain the item. 
+     * @method get
+     * @param key {Map} The key of the item in the map.
+     */
+    this.get = function(key) {
+      return data[key];
+    };
+
+    /***
+     * Adds or updates the item with the given key in the map. 
+     * @method set
+     * @param key The key of the item in the map.
+     * @param value The value to insert into the map of the item in the map.
+     */
+    this.set = function(key, value) {
+      data[key] = value;
+    };
+    
+    /***
+     * Gets a value indicating whether the given key is present in the map.
+     * @method has
+     * @param key The key of the item in the map.
+     * @return {Boolean}
+     */    
+    this.has = function(key) {
+      return key in data;
+    };
+    
+    /***
+     * Removes the item with the given key from the map. 
+     * @method delete
+     * @param key The key of the item in the map.
+     */
+    this.delete = function(key) {
+      delete data[key];
+    };
+  };
+  // Register Map in SlickGrid.
+  // this is done down here after we defined what "Map" will be
+  $.extend(true, window, {
+    "Slick": {
+      "Map": Map,
```

Comment:
any particular reason why this new Map is not part of the other extend on the top
https://github.com/6pac/SlickGrid/blob/94871cbcfb785993d8ba2085927d59945aa8fa52/slick.core.js#L10-L12

I would prefer to see it there all in 1 extend

Reply:
I tried to explain it in the comment down on the place where it is registered. 

The `var Map` is only defined down there, it will be undefined here and therefore "undefined" will be exported. I had it up here at the beginning but got undefined everywhere.

Reply:
or we can move all the other ones down there then, in some files they are defined that way.